#! /usr/bin/env python

PACKAGE='reflexive_plus'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level             Description          Default    Min   Max                                
gen.add("joystickMin",    	double_t, SensorLevels.RECONFIGURE_STOP,    "The minimal value to output as a joystick (slowest speed).",          			0.2,    0.0,  1.0)
gen.add("frontRangeMin",    	double_t, SensorLevels.RECONFIGURE_STOP,    "The average distance (m) the front rangers must read to output joystick min.",     	0.4,    0.01, 6.0)
gen.add("frontRangeMax",     	double_t, SensorLevels.RECONFIGURE_STOP,    "The average distance (m) the front rangers must read to output max joystick value.",       4.0,    0.01, 6.0)
gen.add("sideRangeMin",    	double_t, SensorLevels.RECONFIGURE_STOP,    "The average distance (m) the side rangers must read to output joystick min.",     		0.4,    0.01, 6.0)
gen.add("sideRangeMax",     	double_t, SensorLevels.RECONFIGURE_STOP,    "The average distance (m) the side rangers must read to output max joystick value.",        4.0,    0.01, 6.0)


exit(gen.generate(PACKAGE, "reflexive_speed_limiter.cpp", "SpeedLimit"))